rubrics/impl/src/main/java/org/sakaiproject/rubrics/logic/RubricsServiceImpl.java

```diff
@@ -311,8 +313,10 @@ public void saveRubricAssociation(String tool, String id, Map<String,String> par
                 } else {
                     String input = "{\"toolId\" : \""+tool+"\",\"itemId\" : \"" + id + "\",\"rubricId\" : " + params.get(RubricsConstants.RBCS_LIST) + ",\"metadata\" : {\"created\" : \"" + created + /*"\",\"modified\" : \"" + nowTime +*/ "\",\"ownerId\" : \"" + owner +
 					"\",\"ownerType\" : \"" + ownerType + "\",\"creatorId\" : \"" + creatorId + "\"},\"parameters\" : {" + setConfigurationParameters(params,oldParams) + "}}";
-                    log.debug("Existing association update" + input);
-                    deleteRubricEvaluationsForAssociation(associationHref, tool);
+                    log.debug("Existing association update " + input);
+                    if(Long.valueOf(params.get(RubricsConstants.RBCS_LIST)) != oldRubricId){
```

Comment:
Just use log.debug("Existing association update {}", input) here. May as well defer evaluation.

Reply:
I've a question: If the new rubric id is different to the old (existing?) one, why do we have to delete the existing evaluations before adding the new ones? Or, am I understanding this incorrectly? Could you just explain it a little?

Reply:
Maybe do it on the SAK, the explanation. I find the backend stuff a bit confusing, especially all the non transactional http calls ...